hi-ml-histopathology/src/histopathology/models/deepmil.py

```diff
@@ -308,7 +308,15 @@ def update_results_with_data_specific_info(self, batch: dict, results: dict) ->
                         ResultsKey.TILE_ID: batch[TilesDataset.TILE_ID_COLUMN],
                         ResultsKey.IMAGE_PATH: batch[TilesDataset.PATH_COLUMN]})
 
-        if (TilesDataset.TILE_X_COLUMN in batch.keys()) and (TilesDataset.TILE_Y_COLUMN in batch.keys()):
+        if (TileKey.TILE_TOP in batch.keys()) and (TileKey.TILE_LEFT in batch.keys()) \
+                and (TileKey.TILE_RIGHT in batch.keys()) and (TileKey.TILE_BOTTOM in batch.keys()):
+            results.update({ResultsKey.TILE_TOP: batch[TileKey.TILE_TOP],
+                            ResultsKey.TILE_LEFT: batch[TileKey.TILE_LEFT],
+                            ResultsKey.TILE_RIGHT: batch[TileKey.TILE_RIGHT],
+                            ResultsKey.TILE_BOTTOM: batch[TileKey.TILE_BOTTOM]}
+                           )
+        # the condition below ensures compatibility with older tile datasets
+        elif (TilesDataset.TILE_X_COLUMN in batch.keys()) and (TilesDataset.TILE_Y_COLUMN in batch.keys()):
```

Comment:
In this case, could we not do instead `ResultsKey.TILE_LEFT/TOP: batch[TilesDataset.TILE_X/Y_COLUMN]`? Then we may be able to deprecate/delete `ResultsKey.TILE_X/Y` entirely, remove the `if/else` in `metrics_utils.py` an `test_metrics_utils.py`, and just check for the existence of `ResultsKey.TILE_RIGHT/BOTTOM` where necessary to get the tile size. @fepegar, do you anticipate any issues with this?

Reply:
That sounds like a good idea, but it's hard for me to predict issues happening in other parts of the codebase.

Reply:
@dccastro how should we deal with the `else` when co-ordinates are not available? We will still have to check if `TilesDataset.TILE_X/Y_COLUMN` are available for the dataset, so there will be an if-else

Reply:
@dccastro thinking more about this, we still have default columns as `TilesDataset.TILE_X/Y_COLUMN` and we need to check for co-ordinates in the previously created datasets not having LEFT, TOP etc. , so we need the `elif`. Agreed that we can delete `ResultsKey.TILE_X/Y` for consistent outputs

Reply:
Yes, I agree we still need this `elif` here for the column names; my point was regarding the `ResultsKey`s, which you've already addressed. 🙂